src/abstractdataframe/abstractdataframe.jl

```diff
@@ -405,31 +412,109 @@ describe(df)
 ```
 
 """
-StatsBase.describe(df::AbstractDataFrame) = describe(stdout, df)
-function StatsBase.describe(io, df::AbstractDataFrame)
-    for (name, col) in eachcol(df)
-        println(io, name)
-        describe(io, col)
-        println(io, )
+StatsBase.describe(df::AbstractDataFrame; kwargs...) = describe(stdout, df; kwargs...)
+function StatsBase.describe(io, df::AbstractDataFrame; colstats = [:mean, :min, :median, :max, :Nmissing, :datatype])
+    # Check that people don't specify the wrong fields. 
+    allowed_fields = [:mean, :sd, :min, :q25, :median, :q75, :max, :datatype, :Nunique, :Nmissing]
+    for i in colstats 
+        if !contains(==, allowed_fields, i) 
+            error("""
+                Not an allowed field. Allowed fields are:
+                :mean,
+                :min,
+                :q25,
+                :median,
+                :q75,
+                :max,
+                :datatype,
+                :Nmissing,
+                :Nunique""") 
+        end
     end
-end
-
-function StatsBase.describe(io::IO, X::AbstractVector{Union{T, Missing}}) where T
-    missingcount = count(ismissing, X)
-    pmissing = 100 * missingcount/length(X)
-    if pmissing != 100 && T <: Real
-        show(io, StatsBase.summarystats(collect(skipmissing(X))))
-    else
-        println(io, "Summary Stats:")
+    # Define 4 functions for getting summary statistics 
+    # use a dict because we dont know which measures the user wants
+    function get_stats(col::AbstractArray{T} where T <: Real)
```

Comment:
`AbstractArray{<:Real}` is enough.

Reply:
Thank you, I have updated the type signatures.